homeassistant/components/balboa/climate.py

```diff
@@ -0,0 +1,161 @@
+"""Support for Balboa Spa Wifi adaptor."""
+from __future__ import annotations
+
+import asyncio
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import CLIMATE, CLIMATE_SUPPORTED_FANSTATES, CLIMATE_SUPPORTED_MODES, DOMAIN
+from .entity import BalboaEntity
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the spa climate device."""
+    async_add_entities(
+        [
+            BalboaSpaClimate(
+                hass,
+                entry,
+                hass.data[DOMAIN][entry.entry_id],
+                CLIMATE,
+            )
+        ],
+    )
+
+
+class BalboaSpaClimate(BalboaEntity, ClimateEntity):
+    """Representation of a Balboa Spa Climate device."""
+
+    _attr_icon = "mdi:hot-tub"
+    _attr_fan_modes = CLIMATE_SUPPORTED_FANSTATES
+    _attr_hvac_modes = CLIMATE_SUPPORTED_MODES
+
+    def __init__(self, hass, entry, client, devtype, num=None):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.